tests/unit/test_errorfactory.py

```diff
@@ -0,0 +1,30 @@
+import sys
+
+from nose.tools import assert_equals, assert_true
+
+from botocore.exceptions import ClientError
+from botocore.errorfactory import ServiceErrorFactory
+from botocore.model import ServiceModel
+
+
+def test_errorfactory():
+    model = {
+        'operations': {
+            'OperationName': {
+                'name': 'OperationName',
+                'errors': [{'shape': 'NoSuchResourceException'}],
+            }
+        },
+        'shapes': {
+            'NoSuchResourceException': {
+                'type': 'structure',
+                'members': {}
+            }
+        }
+    }
+    factory = ServiceErrorFactory(ServiceModel(model))
+    members = dir(factory)
+    if sys.version_info >= (2, 7):
```

Comment:
Turns out I do have Python 2.6 (os x comes with it) and here's what I found: Python 2.6 uses an earlier version of the attribute enumeration protocol that second-guesses `__dir__` using (since deprecated) `__members__` and `__methods__` attributes as described in https://www.python.org/dev/peps/pep-0252/. Also, Python 2.6 wants attribute names to be bytestrings and discards them if they are unicode strings (as they are when loaded from the model). Fixed both issues (although IMO Python 2.6 should not be accommodated).